src/Plugin/views/field/HierarchyTreeSummary.php

```diff
@@ -141,32 +85,23 @@ public function query() {
    */
   public function render(ResultRow $values) {
     if ($this->options['summary_type'] == 'child_counts') {
-      $storage = $this->getTreeStorage();
       $output = [];
       if ($entity = $this->getEntity($values)) {
-        $stub = $this->nodeKeyFactory->fromEntity($entity);
-        $level = 1;
-        while ($entities = $storage->findDescendants($stub, 1, $level)) {
-          // This is inefficient and one reason why this is only an admin tool.
-          $entities = $this->treeMapper->loadEntitiesForTreeNodesWithoutAccessChecks($entity->getEntityTypeId(), $entities);
-          $output[] = count($entities);
-          $level++;
+        $queryBuilder = $this->queryBuilderFactory->get($this->configuration['entity_hierarchy_field'], $entity->getEntityTypeId());
```

Comment:
This one in particular used to do N queries per row... where N was the depth of available children. If children were 5 levels deep, it would do 5 queries. So I call it a win :P